jujugui/static/gui/src/app/jujulib/bakery-factory.js

```diff
@@ -0,0 +1,60 @@
+/* Copyright (C) 2016 Canonical Ltd. */
+'use strict';
+
+var module = module;
+
+(function (exports) {
+
+
+  var jujulib = exports.jujulib;
+  /**
+   * BakeryFactory to create and maintain bakery objects.
+   *
+   */
+
+  /**
+   Initialize.
+
+   @method initializer
+   @param {Object} cfg a config object containing the Bakery module
+                   {Bakery: Y.juju.environments.web.Bakery}
+   @return {undefined} Nothing.
+   */
+  function bakeryFactory(cfg) {
+    this.Bakery = cfg.Bakery;
+    this.bakeries = new Map();
```

Comment:
I'd remove bakeries from the public contract if we are not planning to use it directly (and it does not seems we are, since a get method is implemented below). So, `this._bakeries`?